spec/javascripts/modules/cross-domain-tracking.spec.js

```diff
@@ -0,0 +1,97 @@
+describe('Cross Domain Tracking', function () {
+  'use strict'
+  var module
+
+  beforeEach(function () {
+    GOVUK.analytics = GOVUK.analytics || {}
+    GOVUK.analytics.addLinkedTrackerDomain = function () {
+    }
+
+    spyOn(GOVUK.analytics, 'addLinkedTrackerDomain')
+    module = new GOVUK.Modules.CrossDomainTracking()
+  })
+
+  it('tracks realistic example', function () {
+    var anchorToTest = document.createElement('a')
+    anchorToTest.href = 'https://www.registertovote.service.gov.uk/register-to-vote/start'
+    anchorToTest.className = 'button button--start'
+    anchorToTest.setAttribute('role', 'button')
+    anchorToTest.setAttribute('data-module', 'cross-domain-tracking')
+    anchorToTest.setAttribute('data-tracking-code', 'UA-23066786-5')
+    anchorToTest.setAttribute('data-tracking-name', 'transactionTracker')
+    anchorToTest.textContent = 'Start Now'
```

Comment:
Would this be easier to read if the test anchor was setup as a simple string, like:
`"<a href="https://www.registertovote.service.gov.uk/register-to-vote/start" …>Start now</a>"`

Reply:
These tests do not pass if given a string, I think trying to get the 'hostname' of a link in a jQuery fragment does not work since it's not been created in a real dom.

I reference this in the commit message https://github.com/alphagov/static/pull/1193/commits/de832f6bd625e53de4f5cf54074a6ef666f03f4d